test/e2e/specs/editor/various/shortcut-focus-toolbar.spec.js

```diff
@@ -0,0 +1,269 @@
+	},
+} );
+
+test.describe( 'Focus toolbar shortcut (alt + F10)', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'Focuses correct toolbar in default view options in edit mode', async ( {
+		editor,
+		page,
+		toolbarUtils,
+	} ) => {
+		// Test: Focus the top level toolbar from title
+		await toolbarUtils.moveToToolbarShortcut();
+		await expect( toolbarUtils.documentToolbarButton ).toBeFocused();
+
+		// Test: Focus document toolbar from empty block
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await toolbarUtils.moveToToolbarShortcut();
+		await expect( toolbarUtils.documentToolbarButton ).toBeFocused();
+
+		// Test: Focus block toolbar from block content when block toolbar isn't visible
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await page.keyboard.type(
+			"Focus to block toolbar when block toolbar isn't visible"
+		);
+		await toolbarUtils.moveToToolbarShortcut();
+		await expect( toolbarUtils.blockToolbarParagraphButton ).toBeFocused();
+		await expect( toolbarUtils.documentToolbarTooltip ).not.toBeVisible();
+
+		// Test: Focus block toolbar from block content when block toolbar is visible
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await page.keyboard.type(
+			'Focus to block toolbar when block toolbar is visible'
+		);
+		// We need to force the toolbar to show. Otherwise, the bug from
+		// https://github.com/WordPress/gutenberg/pull/49644 won't surface in the e2e tests.
+		await editor.showBlockToolbar();
+		await toolbarUtils.moveToToolbarShortcut();
+		await expect( toolbarUtils.blockToolbarParagraphButton ).toBeFocused();
+		await expect( toolbarUtils.documentToolbarTooltip ).not.toBeVisible();
+	} );
+
+	test( 'Focuses correct toolbar in default view options in select mode', async ( {
+		editor,
+		page,
+		toolbarUtils,
+	} ) => {
+		// Test: Focus the document toolbar from title
+		await toolbarUtils.useSelectMode();
+		await toolbarUtils.moveToToolbarShortcut();
+		await expect( toolbarUtils.documentToolbarButton ).toBeFocused();
+
+		// Test: Focus the top level toolbar from empty block
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await toolbarUtils.useSelectMode();
+		await toolbarUtils.moveToToolbarShortcut();
+		await expect( toolbarUtils.documentToolbarButton ).toBeFocused();
+
+		// Test: Focus the top level toolbar from paragraph block
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await page.keyboard.type(
+			'Focus top level toolbar from paragraph block in select mode.'
+		);
+		await toolbarUtils.useSelectMode();
+		await toolbarUtils.moveToToolbarShortcut();
+		await expect( toolbarUtils.documentToolbarButton ).toBeFocused();
+	} );
+
+	test.describe( 'In Top Toolbar option:', () => {
+		test.beforeEach( async ( { editor } ) => {
+			// Ensure the fixed toolbar option is on
+			await editor.setIsFixedToolbar( true );
+		} );
```

Comment:
@jeryj, the block editor settings are persisted across the tests, and this setting occasionally might leak to other tests. We should probably add cleanup `test.afterEach` and restore the toolbar setting.

I noticed while debugging #49733.

## Screenshot from the test failure
![test-failed-1](https://github.com/WordPress/gutenberg/assets/240569/0c1bb2da-d043-40f1-af4c-c79566d2f527)

Reply:
Nice catch! Put in a fix here: https://github.com/WordPress/gutenberg/pull/51600